javascript/ql/lib/semmle/javascript/dataflow/internal/InterProceduralTypeInference.qll

```diff
@@ -190,16 +190,32 @@ private VarAccess getOnlyAccess(FunctionDeclStmt fn, LocalVariable v) {
   result = unique(VarAccess acc | acc = v.getAnAccess())
 }
 
+private VarAccess getOnlyAccessToFunctionExpr(FunctionExpr fn, LocalVariable v) {
+  exists( DeclStmt st |
+    fn = st.(DeclStmt).getADecl().getInit() and 
+    v = st.(DeclStmt).getADecl().getBindingPattern().getVariable() and
+    result = unique(VarAccess acc | acc = v.getAnAccess())
+  )
+}
+
 /** A function that only is used locally, making it amenable to type inference. */
 class LocalFunction extends Function {
   DataFlow::Impl::ExplicitInvokeNode invk;
 
   LocalFunction() {
-    exists(LocalVariable v |
-      getOnlyAccess(this, v) = invk.getCalleeNode().asExpr() and
-      not exists(v.getAnAssignedExpr()) and
-      not exists(ExportDeclaration export | export.exportsAs(v, _))
-    ) and
+    (
+      exists(LocalVariable v |
+        getOnlyAccess(this, v) = invk.getCalleeNode().asExpr() and
+        not exists(v.getAnAssignedExpr()) and
+        not exists(ExportDeclaration export | export.exportsAs(v, _))
+      )
+    or
+      exists(LocalVariable v |
+        getOnlyAccessToFunctionExpr(this, v) = invk.getCalleeNode().asExpr() and
+        not exists(ExportDeclaration export | export.exportsAs(v, _))
+      )
+    )
+    and
```

Comment:
```suggestion
    exists(LocalVariable v |
      getOnlyAccess(this, v) = invk.getCalleeNode().asExpr() and
      not exists(v.getAnAssignedExpr()) and
      not exists(ExportDeclaration export | export.exportsAs(v, _))
      or
      getOnlyAccessToFunctionExpr(this, v) = invk.getCalleeNode().asExpr() and
      not exists(ExportDeclaration export | export.exportsAs(v, _))
    ) and
```

My suggestion here is purely stylistic and I haven't changed the behavior.  

It's up to you whether you want to use my suggestion, but if you don't, then you have to run the autoformatter.